activesupport/lib/active_support/time_with_zone.rb

```diff
@@ -303,6 +303,13 @@ def +(other)
         result = utc + other
         result.in_time_zone(time_zone)
       end
+    rescue TypeError
+      ActiveSupport.deprecator.warn(
+        "Do not add an instance of #{other.class} to an instance of #{utc.class}. This behaviour will raise " \
+        "in Rails 8.0."
```

Comment:
We've standardized Rails on US english. Generally deprecations are also written as what is deprecated, rather than telling someone not to do something.

```suggestion
        "Adding an instance of #{other.class} to an instance of #{utc.class} is deprecated. This behavior will raise " \
        "a `TypeError` in Rails 8.0."
```